src/Statement/StatementList.php

```diff
@@ -45,10 +45,20 @@ public function __construct( $statements = array() /*...*/ ) {
 			$statements = func_get_args();
 		}
 
+		$this->setStatements( $statements );
+	}
+
+	/**
+	 * @param Statement[]|Traversable
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function setStatements( $statements ) {
 		if ( !is_array( $statements ) && !( $statements instanceof Traversable ) ) {
 			throw new InvalidArgumentException( '$statements must be an array or an instance of Traversable' );
 		}
 
+		$this->statements = array();
```

Comment:
That is a good point indeed. Perhaps having a local variable here in the first place and setting it to the field only at the end of the method would be better.